liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/applications/LooseEarApplication.java

```diff
@@ -0,0 +1,145 @@
+        return jarArchive;
+    }
+    
+    public Element addEjbModule(MavenProject proj) throws IOException {
+        String ejbArchiveName = "/" + getModuleName(proj);
+        Element ejbArchive = config.addArchive(ejbArchiveName);
+        config.addDir(ejbArchive, proj.getBasedir().getCanonicalPath() + "/target/classes", "/");
+        config.addDir(ejbArchive, proj.getBasedir().getCanonicalPath() + "/src/main/resources", "/");
+        return ejbArchive;
+    }
+    
+    public Element addWarModule(MavenProject proj, String warSourceDir) throws IOException {
+        String warArchiveName = "/" + getModuleName(proj);
+        Element warArchive = config.addArchive(warArchiveName);
+        config.addDir(warSourceDir, "/");
+        config.addDir(warArchive, proj.getBasedir().getCanonicalPath() + "/target/classes", "/WEB-INF/classes");
+        config.addDir(warArchive, proj.getBasedir().getCanonicalPath() + "/src/main/resources", "/WEB-INF/classes");
+        
+        return warArchive;
+    }
+    
+    public void addModuleFromM2(Artifact artifact, String artifactFile) {
+        String artifactName = "/" + getModuleName(artifact);
+        
+        if (getEarDefaultLibBundleDir() != null && (artifact.getType() == null || "jar".equals(artifact.getType()))) {
+            artifactName = "/" + getEarDefaultLibBundleDir() + artifactName;
+        }
+        
+        config.addFile(artifactFile, artifactName);
+    }
+    
+    public String getModuleName(Artifact artifact) {
+        String moduleName;
+        
+        String fileExtension = artifact.getType();
+        if ("ejb".equals(fileExtension)) {
+            fileExtension = "jar";
+        }
+        
+        switch (getEarFileNameMapping()) {
+            case "no-version":
+                moduleName = artifact.getArtifactId() + "." + fileExtension;
+                break;
+            case "no-version-for-ejb":
+                if ("ejb".equals(artifact.getType())) {
+                    moduleName = artifact.getArtifactId() + "." + fileExtension;
+                } else {
+                    moduleName = artifact.getArtifactId() + "-" + artifact.getVersion() + "." + fileExtension;
+                }
+                break;
+            case "full":
+                moduleName = artifact.getGroupId() + "-" + artifact.getArtifactId() + "-" + artifact.getVersion() + "."
+                        + fileExtension;
+                break;
+            default:
+                // standard
+                moduleName = artifact.getArtifactId() + "-" + artifact.getVersion() + "." + fileExtension;
+        }
+        return moduleName;
+    }
+    
+    public String getModuleName(MavenProject proj) {
+        String moduleName;
+        
+        String fileExtension = proj.getPackaging();
+        if ("ejb".equals(fileExtension)) {
+            fileExtension = "jar";
+        }
+        
+        switch (getEarFileNameMapping()) {
```

Comment:
Refactor this switch into a method that takes parameters for the group, name, version, type, etc. to be reused here and from line 69.